projects/storefrontapp-e2e-cypress/cypress/integration/regression/pickup-in-store/pickup-checkout-journey.core-e2e-spec.ts

```diff
@@ -0,0 +1,170 @@
+import {
+  defaultAddress,
+  defaultPaymentDetails,
+  EMAIL_ADDRESS,
+  fillAddressForm,
+  fillPaymentForm,
+  isSorted,
+  LOCATORS as L,
+  login,
+  mockLocation,
+  register,
+} from '../../../helpers/pickup-in-store-utils';
+import { viewportContext } from '../../../helpers/viewport-context';
+
+/*
+- A guest user navigates to a PDP wishing to buy the product.
+- The user has the choice of whether they want the product delivered (the default) or whether they want to pick it up in store.
+- The user selects pickup in store.
+- The user selects which store they want to collect from (by default the last store they selected, falling back to the nearest store).
+- The user adds the product to the cart. (The cart entries post call will have the "deliveryPointOfService" field).
+- From the cart, the user can change the location they wish to pick up the product from.
+- The user also add another item only for delivery.(Multiple items in cart)
+- The user decides to login so the order will show in the user's account.
+- The logged in user checks out.
+TODO:- During checkout, the user can change the pickup location. (WIP)
+- During the order review, the user can change the pickup location.
+TODO:- The user completes checkout and sees the order details. On here they can see their pickup location for the picket item and the delivery for the delivered item. (WIP)
+*/
+
+describe('Pickup Delivery Option', () => {
+  viewportContext(['desktop'], () => {
+    beforeEach(() => {
+      cy.window().then((win) => win.sessionStorage.clear());
+      cy.cxConfig({
+        context: {
+          baseSite: ['apparel-uk-spa'],
+          currency: ['GBP'],
+        },
+      });
+      cy.visit('/', mockLocation(53, 0));
+      cy.get(L.ALLOW_COOKIES_BUTTON).click();
+    });
+
+    it('A user who has a cart with multiple entries checkout with BOPIS', () => {
+      cy.get(L.HOME_PAGE_FIRST_PRODUCT).click();
+      cy.get(L.PICKUP_OPTIONS_RADIO_PICKUP).should('be.visible');
+      cy.get(L.PICKUP_OPTIONS_RADIO_DELIVERY).should('be.visible');
+      cy.get(L.PICKUP_OPTIONS_RADIO_DELIVERY).should('be.checked');
+
+      cy.get(L.PICKUP_OPTIONS_RADIO_PICKUP).click();
+      cy.get(L.PICKUP_OPTIONS_RADIO_PICKUP).should('be.checked');
+      cy.get(L.USE_MY_LOCATION).click();
+      cy.get(L.STORE_DISTANCE).then(($distance) => {
+        let distanceArr = $distance.text().split(' ');
+        expect(isSorted(distanceArr)).to.be.true;
+      });
+      cy.get(L.ACTIVE_PICK_UP_IN_STORE_BUTTON).first().click();
+
+      cy.get(L.ADD_TO_CART).click();
+      cy.intercept({
+        method: 'POST',
+        url: /users\/anonymous\/carts\/[0-9| a-z|-]*\/entries/,
+      }).as('apiAddToCart');
```

Comment:
Move interceptors above the actions that can trigger them to avoid race conditions